src/stacking.rs

```diff
@@ -76,26 +73,45 @@ where
     Ok(res)
 }
 
-/// Stack arrays along the given axis.
+pub fn stack<A, D>(
+    axis: Axis,
+    arrays: Vec<ArrayView<A, D>>,
+) -> Result<Array<A, D::Larger>, ShapeError>
+where
+    A: Copy,
+    D: Dimension,
+    D::Larger: RemoveAxis,
+{
+    if let Some(ndim) = D::NDIM {
+        if axis.index() > ndim {
+            return Err(from_kind(ErrorKind::OutOfBounds));
+        }
+    }
+    let arrays: Vec<ArrayView<A, D::Larger>> =
+        arrays.into_iter().map(|a| a.insert_axis(axis)).collect();
+    concatenate(axis, &arrays)
```

Comment:
Implementation strategy wise, can't we take the old `stack`, put that into an internal function. Give it a parameter that tells it whether to remove an axis or not. It can be a type parameter, since it influences the return type (the dim of the returned array).

That way both stack and concatenate can use this internal function and we don't need the intermediate vector here in concatenate

Reply:
Looks good! I'll try to implement it this way.